src/script/page/RightSidebar/MessageDetails/MessageDetails.tsx

```diff
@@ -119,15 +134,28 @@ const MessageDetails: FC<MessageDetailsProps> = ({
       return receiptUsers.length ? MESSAGE_STATES.RECEIPTS : MESSAGE_STATES.NO_RECEIPTS;
     }
 
-    return likeUsers.length ? MESSAGE_STATES.LIKES : MESSAGE_STATES.NO_LIKES;
-  }, [supportsReceipts, isReceiptsOpen, messageEntity, receiptUsers, likeUsers]);
-
-  const getLikes = useCallback(async (reactions: UserReactionMap) => {
-    const currentLikes = Object.keys(reactions);
-    const usersLikes = await userRepository.getUsersById(currentLikes.map(likeId => ({domain: '', id: likeId})));
-    const sortedUsersLikes = usersLikes.sort(sortUsers);
+    return getTotalReactionUsersCount(reactionUsers) ? MESSAGE_STATES.REACTIONS : MESSAGE_STATES.NO_REACTIONS;
+  }, [supportsReceipts, isReceiptsOpen, messageEntity, receiptUsers, reactionUsers]);
+
+  const getReactions = useCallback(async (reactions: UserReactionMap) => {
+    const usersMap = new Map<string, User>();
+    const currentReactions = Object.keys(reactions);
+    const usersReactions = await userRepository.getUsersById(
+      currentReactions.map(userId => ({domain: '', id: userId})),
+    );
+    usersReactions.forEach(user => {
+      usersMap.set(user.id, user);
+    });
+    const reactionsGroupByUser = groupByReactionUsers(reactions);
+    const reactionsGroupByUserMap = new Map<string, User[]>();
+    reactionsGroupByUser.forEach((userIds, reaction) => {
+      reactionsGroupByUserMap.set(
+        reaction,
+        userIds.map(userId => usersMap.get(userId)!),
+      );
+    });
 
-    setLikeUsers(sortedUsersLikes);
+    setReactionUsers(reactionsGroupByUserMap);
   }, []);
```

Comment:
LEt's remove the `useCallback` here, it unnecessary complexity and add more computation while rendering. 
Callbacks do not affect rendering performances

Reply:
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/63250054/226327066-8e692973-6025-4757-969e-0442ed61d7c2.png">